build.gradle

```diff
@@ -71,46 +71,37 @@ allprojects {
                 '-Xlint:deprecation',
                 '-Xlint:unchecked',
             ]
-            options.errorprone.disableWarningsInGeneratedCode = true
-            options.errorprone.errorproneArgs.addAll([
-                '-Werror',
-                '-Xlint:deprecation',
-                '-Xlint:unchecked',
-                '-XepDisableWarningsInGeneratedCode',
-                '-XepExcludedPaths:.*(generated).*',
-                '-XepOpt:NullAway:AnnotatedPackages=com.palantir',
-            ])
-
-            // warnings not explicitly provided by error-prone
-            Set<String> warningsToPromote = [
-                'NullAway',
-                'Slf4jLogsafeArgs',
-                'PreferCollectionTransform',
-                'PreferListsPartition',
-                'PreferSafeLoggingPreconditions',
-                'PreferSafeLoggableExceptions',
-            ]
-
-            com.google.errorprone.scanner.BuiltInCheckerSuppliers.ENABLED_WARNINGS.forEach {
-                check -> warningsToPromote << check.canonicalName()
-            }
 
-            com.google.errorprone.scanner.BuiltInCheckerSuppliers.DISABLED_CHECKS.forEach {
-                check -> warningsToPromote << check.canonicalName()
+            options.errorprone {
+                disableWarningsInGeneratedCode = true
+                excludedPaths = '.*(generated).*'
+
+                errorproneArgs.addAll([
+                    '-Werror',
+                    '-Xlint:deprecation',
+                    '-Xlint:unchecked',
+                ])
```

Comment:
I wouldn't expect these to have any impact, shouldn't error-prone inherit them from the java compiler settings?
excludedPaths and disableWarningsInGeneratedCode should be inherited from baseline, however I think the `excludedPaths` here is a little broader than the default.

Reply:
I seem to recall there was some point in time where the `options.compilerArgs` didn't get inherited and we weren't actually catching some compiler warnings as errors so I will need to test adding in an explicit break to test that would fire if we remove these.

Reply:
updated, did a quick test locally after removing these and error-prone compiler properly flagged an `unchecked` warning and treated it as a compilation error.

Reply:
I'd imagine these could be removed as well:
```
disableWarningsInGeneratedCode = true
excludedPaths = '.*(generated).*'
```